### PR TITLE
Release 1.0.0

### PR DESCRIPTION
**Pull request by bcardiff**
Preview at https://github.com/bcardiff/crystal/blob/changelog/1.0.0/CHANGELOG.md

✨✨✨

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,106 @@
...
+- Fix `Range#==` to ignore generic type arguments. ([#10309](https://github.com/crystal-lang/crystal/pull/10309), thanks @straight-shoota)
+- Make `Enumerable#flat_map`, `Iterator#flat_map` work with mixed element types. ([#10329](https://github.com/crystal-lang/crystal/pull/10329), thanks @HertzDevil)
+- Remove duplicated `sort` related specs. ([#10208](https://github.com/crystal-lang/crystal/pull/10208), thanks @MakeNowJust)
+- Fix docs regarding `Set#each` return type. ([#10477](https://github.com/crystal-lang/crystal/pull/10477), thanks @kachick)
+- Fix docs examples regarding `Set#*set_of?`. ([#10285](https://github.com/crystal-lang/crystal/pull/10285), thanks @oddp)
+- Fix expectation on set specs. ([#10482](https://github.com/crystal-lang/crystal/pull/10482), thanks @kachick)
+
+### Serialization
+
+- **(breaking-change)** Serialize `Enum` to underscored `String` by default. ([#10431](https://github.com/crystal-lang/crystal/pull/10431), thanks @straight-shoota)
```
Comment (caspiano) [inline CHANGELOG.md:40]: And me? 🥺👉👈

```suggestion
- **(breaking-change)** Serialize `Enum` to underscored `String` by default. ([#10431](https://github.com/crystal-lang/crystal/pull/10431), thanks @straight-shoota, @caspiano)
```

**Comment by bcardiff** [comment]
This deserves some gifs. Behold the `:crystalparrot:`



credits: @pyritewolf

File: CHANGELOG.md (hunk 1 above)
Comment (bcardiff) [reply CHANGELOG.md]: Totally!

**Comment by sdogruyol** [comment]
OMG! This is it 💯 

Been waiting for this since 2014, now I can finally say that one of my dreams came true 😄 

Thanks to everyone who supported and helped Crystal in any way 🙏  
You rock 🤘

**Comment by Fryguy** [comment]
:shipit:

**Comment by wontruefree** [comment]
I am so excited for this.  It is the biggest hurdle for adoption at work.

**Comment by LeandroRezendeCoutinho** [comment]
Awesome! Long time waiting... Nice job.

**Comment by vlazar** [comment]
Not complaining or something, but wasn't the plan to have several `1.0.0-pre` first for smooth `1.0.0`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,106 @@
...
+- Add `HTTP::Request#local_address`. ([#10385](https://github.com/crystal-lang/crystal/pull/10385), thanks @carlhoerberg)
+
+### Logging
+
+- Close `AsyncDispatcher` on `#finalize`. ([#10390](https://github.com/crystal-lang/crystal/pull/10390), thanks @straight-shoota)
+
+### System
+
+- Fix `Process.parse_argument` behavior against a quote in a word. ([#10337](https://github.com/crystal-lang/crystal/pull/10337), thanks @MakeNowJust)
+- Add aarch64 support for macOS/darwin targets. ([#10348](https://github.com/crystal-lang/crystal/pull/10348), thanks @maxfierke)
```
Comment (maxfierke) [inline CHANGELOG.md:68]: @RomainFranceschini deserves a call-out here too 😃

**Comment by EPonyA** [comment]
> I am so excited for this. It is the biggest hurdle for adoption at work.

That seems strange to me. 1.0 is a social construct, not a real guarantee that it's any more or less production ready than 0.1. There are still known bugs, and every 1.0 lang I'm aware of had bugs on release. But Idk.

**Comment by luislavena** [comment]
> That seems strange to me. 1.0 is a social construct, not a real guarantee that it's any more or less production ready than 0.1

That is correct, however certain company policies might say no production system should be built on a language or tool that hasn't reached out 1.0, which can me immediately refuted by stating that huge projects like React Native and many others are still in the 0.x versions and constantly used in production ([source](https://0ver.org/)). 🤷🏽 

Policies, fear of the unknown, whatever, I'm just happy to see Crystal putting the 1.0 🏷️(label) 😄 

Congratulations Crystal team! 🎉 🥳 👏🏽

File: CHANGELOG.md (hunk 2 above)
Comment (RomainFranceschini) [reply CHANGELOG.md:68]: Thanks @maxfierke 😄

Comment (RomainFranceschini) [inline CHANGELOG.md:68]:
```suggestion
- Add aarch64 support for macOS/darwin targets. ([#10348](https://github.com/crystal-lang/crystal/pull/10348), thanks @maxfierke, @RomainFranceschini)
```

**Comment by HertzDevil** [comment]
Do we need to change the [project status](https://github.com/crystal-lang/crystal/blob/7aa9e98a195a2da63f703be36e5694074a1179a9/README.md#project-status) too? (especially the part about breaking changes)

**Comment by jwaldrip** [comment]
Will this ship with Apple silicon support?

**Comment by bcardiff** [comment]
@jwaldrip There will be no binary distribution for Apple silicon for 1.0. Its support and instructions to compiler a native aarch64 darwin compiler are detailed in #10348

**Comment by 636f7374** [comment]
Great, I have been using Crystal since 2019!

**Review by RX14 — APPROVED** [review]
Congrats! :tada: 

This is well-deserved!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,116 @@
...
+- Fix ICE when exhaustive in-clause calls pseudo-method. ([#10382](https://github.com/crystal-lang/crystal/pull/10382), thanks @HertzDevil)
+- Fix ICE when parsing `foo.%` calls. ([#10351](https://github.com/crystal-lang/crystal/pull/10351), thanks @MakeNowJust)
+- Fix edge cases for symbol quoting rules. ([#10389](https://github.com/crystal-lang/crystal/pull/10389), thanks @HertzDevil)
+- Support closured vars inside `Const` initializer. ([#10478](https://github.com/crystal-lang/crystal/pull/10478), thanks @RX14)
+- Disallow surrogate halves in escape sequences of string and character literals. ([#10443](https://github.com/crystal-lang/crystal/pull/10443), thanks @HertzDevil)
+- Documentation grammar fix. ([#10369](https://github.com/crystal-lang/crystal/pull/10369), thanks @szTheory)
+
+### Language semantics
+
+- Don't fail on untyped `is_a`. ([#10320](https://github.com/crystal-lang/crystal/pull/10320), thanks @asterite)
```
Comment (straight-shoota) [inline CHANGELOG.md:90]:
```suggestion
- Don't fail on untyped `is_a?`. ([#10320](https://github.com/crystal-lang/crystal/pull/10320), thanks @asterite)
```

**Review by straight-shoota — APPROVED** [review]
I'd prefer to release 1.0.0-pre first, but the changelog looks fine except for the small typo.

**Review by straight-shoota — APPROVED** [review]

**Comment by pirj** [comment]
Congrats on the release! Let the crystal shine! ❤️

**Comment by straight-shoota** [comment]
Release post:

https://crystal-lang.org/2021/03/22/crystal-1.0-what-to-expect.html

**Comment by marcoroth** [comment]
Congrats on the 1.0.0 release! 🚀 🎉

**Comment by 636f7374** [comment]
Congratulations, hope it will be better next!

**Comment by kachick** [comment]
Congratulations! 🎉

**Comment by defp** [comment]
🎉🎉🎉

**Pull request merged**